src/django_upgrade/fixers/test_http_headers.py

```diff
@@ -0,0 +1,189 @@
+            tokens, i, kwarg_name=http_kwarg_name
+        )
+        new_headers_kwarg_values[transform_header_arg(http_kwarg_name)] = erased_tokens
+        if first_start_idx == -1:
+            first_start_idx = start_idx
+
+    if new_headers_kwarg_values:
+        if contains_headers_kwarg:
+            insert_into_existing_headers_kwarg(
+                tokens, i, kwarg_dict=new_headers_kwarg_values
+            )
+        else:
+            insert_headers_kwarg(
+                tokens,
+                i,
+                kwarg_dict=new_headers_kwarg_values,
+                insert_position=first_start_idx,
+            )
+
+
+def transform_header_arg(header: str) -> str:
+    return str_removeprefix(header, HTTP_PREFIX).replace("_", "-").lower()
+
+
+def erase_keyword_argument(
+    tokens: list[Token], i: int, *, kwarg_name: str
+) -> tuple[list[Token], int]:
+    """
+    Erases a keyword argument based on its name, and returns the tokens that
+    defined the value of the keyword argument, and its starting position.
+    The starting position will be used to start inserting the new argument in
+    the right place.
+    """
+    open_idx = find(tokens, i, name=OP, src="(")
+    func_args, close_idx = parse_call_args(tokens, open_idx)
+    for arg_start_idx, arg_end_idx in func_args:
+        kwarg_name_start_idx = find(tokens, arg_start_idx, name=NAME, src=kwarg_name)
+        if kwarg_name_start_idx > arg_end_idx:
+            continue
+
+        value_start_idx = find(tokens, kwarg_name_start_idx, name=OP, src="=")
+        erased_tokens = tokens[value_start_idx + 1 : arg_end_idx]
+
+        ends_with_comma = (
+            tokens[arg_end_idx].name == OP and tokens[arg_end_idx].src == ","
+        )
+        if ends_with_comma:
+            arg_end_idx += 1
+        ends_with_whitespace = (
+            tokens[arg_end_idx].name == "UNIMPORTANT_WS"
+            and tokens[arg_end_idx].src == " "
+        )
+        if ends_with_whitespace:
+            arg_end_idx += 1
+
+        del tokens[kwarg_name_start_idx:arg_end_idx]
+        return erased_tokens, kwarg_name_start_idx
+    return [], -1
+
+
+def insert_into_existing_headers_kwarg(
+    tokens: list[Token], i: int, *, kwarg_dict: dict[str, list[Token]]
+) -> None:
+    open_idx = find(tokens, i, name=OP, src="(")
+
+    headers_arg_start_idx = find(tokens, open_idx, name=NAME, src=HEADERS_KWARG)
+    headers_closing_idx = find(tokens, headers_arg_start_idx, name=OP, src="}")
+
+    new_tokens = []
+    for key, value in kwarg_dict.items():
+        new_tokens += [
+            Token(name="OP", src=","),
+            Token(name="UNIMPORTANT_WS", src=" "),
+        ]
+        new_tokens += [
+            Token(name="STRING", src=f'"{key}"'),
+            Token(name="OP", src=":"),
+            Token(name="UNIMPORTANT_WS", src=" "),
+        ]
+        new_tokens += value
+    tokens[headers_closing_idx:headers_closing_idx] = new_tokens
```

Comment:
when injecting new code we don't need to build up loads of individual tokens, we can just create a single token with the target source.